src/bound_propagation/polynomial.py

```diff
@@ -310,10 +310,16 @@ def __init__(self, monomials):
 
             monomial_indices.append(monomial_index)
 
-        super().__init__(
-            Parallel(Select(linear_terms), UnivariateMonomial(factors)),
-            Parallel(*[self.construct_mul(monomial_index) for monomial_index in monomial_indices])
-        )
+        if factors:
+            modules = [
```

Comment:
This branch still ought to have the multiplication of monomials (self.construct_mul for multiplication of univariate monomials with different variables). Otherwise, the output will just be a list of univariate monomials. :-)

Reply:
Of course, typo!